enterprise/frontend/src/metabase-enterprise/transforms/components/PythonTransformEditor/PythonDataPicker/PythonDataPicker.tsx

```diff
@@ -0,0 +1,285 @@
+  tables: PythonTransformTableAliases;
+  onChange: (
+    database: number,
+    tables: PythonTransformTableAliases,
+    tableInfo: Table[],
+  ) => void;
+};
+
+export function PythonDataPicker({
+  database,
+  tables,
+  onChange,
+}: PythonDataPickerProps) {
+  const [tableSelections, setTableSelections] = useState<TableSelection[]>(
+    getInitialTableSelections(tables),
+  );
+
+  const {
+    data: databases,
+    isLoading: isLoadingDatabases,
+    error: databaseError,
+  } = useListDatabasesQuery();
+
+  const {
+    data: tablesData,
+    isLoading: isLoadingTables,
+    error: tablesError,
+  } = useListTablesQuery(
+    database
+      ? {
+          dbId: database,
+          include_hidden: false,
+          include_editable_data_model: true,
+        }
+      : skipToken,
+  );
+
+  const handleChange = (
+    database: DatabaseId | undefined,
+    selections: TableSelection[],
+  ) => {
+    if (database) {
+      const tableAliases = selectionsToTableAliases(selections);
+      onChange(database, tableAliases, tablesData ?? []);
+    }
+  };
+
+  if (databaseError || tablesError) {
+    return <LoadingAndErrorWrapper error={databaseError || tablesError} />;
+  }
+
+  if (isLoadingDatabases) {
+    return <LoadingAndErrorWrapper loading />;
+  }
+
+  const availableTables = (tablesData || []).filter(
+    (tbl: Table) => tbl.db_id === database && tbl.active,
+  );
+
+  const usedAliases = new Set(tableSelections.map((s) => s.alias));
+
+  const handleDatabaseChange = (value: string | null) => {
+    const newDatabase = value ? parseInt(value) : undefined;
+
+    // Clear selections since they won't make sense
+    const newSelections = [
+      {
+        tableId: undefined,
+        alias: "",
+      },
+    ];
+    setTableSelections(newSelections);
+    handleChange(newDatabase, newSelections);
+  };
+
+  const handleSelectionChange = (
+    selectionIndex: number,
+    selection: TableSelection,
+  ) => {
+    const newSelections = Array.from(tableSelections);
+    newSelections[selectionIndex] = selection;
+    setTableSelections(newSelections);
+    handleChange(database, newSelections);
+  };
+
+  const handleAddTable = () => {
+    setTableSelections([
+      ...tableSelections,
+      {
+        tableId: undefined,
+        alias: "",
+      },
+    ]);
+  };
+
+  const handleRemoveTable = (selectionIndex: number) => {
+    const newSelections = Array.from(tableSelections);
+    newSelections.splice(selectionIndex, 1);
+    setTableSelections(newSelections);
+    handleChange(database, newSelections);
+  };
+
+  return (
+    <Stack
+      p="md"
+      gap="md"
+      h="100%"
+      w="30%"
+      miw="300px"
+      className={S.dataPicker}
+      data-testid="python-data-picker"
+    >
+      <Box>
+        <Text fw="bold">{t`Source database`}</Text>
+        <Text size="sm" c="text-light" mb="sm">
+          {t`Select the database that contains your source data`}
+        </Text>
+
+        <DatabaseDataSelector
```

Comment:
This picker is just temporary until we have multi-db Python transforms, it will go away eventually.